dashboard/app/assets/javascripts/header.js.erb

```diff
@@ -413,3 +416,47 @@ dashboard.header.updateTimestamp = function () {
     $('.project_updated_at').text("Not saved"); // TODO i18n
   }
 };
+
+function getSummarizedProgressForAnonymousUser () {
+  var summarizedProgress = {};
+  var levelProgress = {};
+  var cookie = document.cookie.split(';');
+  
+  for(var i = 0; i < cookie.length; i++) {
+    var pair = cookie[i].split('=');
+    var key = pair[0].trim();
+
+    switch(key) {
+      case 'progress':
+        var allLevelsProgress = JSON.parse(decodeURIComponent(pair[1]));
+        for(var level in allLevelsProgress) {
+          if (allLevelsProgress.hasOwnProperty(level)) {
+            var individualLevelProgressDescription = '';
```

Comment:
Consider adding a getLevelProgressDescription(progress) helper for readability and testability.

We tend to be a bit more concise in our variable names;  I think just levelProgressDescription is perfectly clear in this context.